services/loadbalancer/wait.go

```diff
@@ -0,0 +1,121 @@
+package loadbalancer
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/stackitcloud/stackit-sdk-go/core/utils"
+	"github.com/stackitcloud/stackit-sdk-go/core/wait"
+)
+
+// Load balancer instance status
+const (
+	InstanceStatusUnspecified = "STATUS_UNSPECIFIED"
+	InstanceStatusPending     = "STATUS_PENDING"
+	InstanceStatusReady       = "STATUS_READY"
+	InstanceStatusError       = "STATUS_ERROR"
+	InstanceStatusTerminating = "STATUS_TERMINATING"
+)
+
+// Load balancing functionality status
+const (
+	FunctionalityStatusUnspecified = "STATUS_UNSPECIFIED"
+	FunctionalityStatusReady       = "STATUS_READY"
+	FunctionalityStatusFailed      = "STATUS_FAILED"
+	FunctionalityStatusUpdating    = "STATUS_UPDATING"
+	FunctionalityStatusDeleting    = "STATUS_DELETING"
+	FunctionalityStatusDisabled    = "STATUS_DISABLED"
+)
+
+// Interface needed for tests
+type APIClientInterface interface {
+	GetLoadBalancerExecute(ctx context.Context, projectId, name string) (*LoadBalancer, error)
+	GetStatusExecute(ctx context.Context, projectId string) (*StatusResponse, error)
+}
+
+func handleError(reqErr error) (res interface{}, done bool, err error) {
+	oapiErr, ok := reqErr.(*GenericOpenAPIError) //nolint:errorlint //complaining that error.As should be used to catch wrapped errors, but this error should not be wrapped
+	if !ok {
+		return nil, false, fmt.Errorf("could not convert error to GenericOpenApiError")
+	}
+	// Some APIs may return temporary errors and the request should be retried
+	if utils.Contains(wait.RetryHttpErrorStatusCodes, oapiErr.statusCode) {
+		return nil, false, nil
+	}
+	return nil, false, reqErr
+}
+
+// CreateInstanceWaitHandler will wait for creation
+func CreateInstanceWaitHandler(ctx context.Context, a APIClientInterface, projectId, instanceName string) *wait.Handler {
```

Comment:
No waiter needed for the UpdateTargetPools?

Reply:
Discussed offline, will confirm with team and add later if needed